addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/FullPathDisclosureScanRule.java

```diff
@@ -0,0 +1,120 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.List;
+import java.util.Map;
+import com.google.re2j.Pattern;
+import com.google.re2j.Matcher;
+import net.htmlparser.jericho.Source;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.CommonAlertTag;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class FullPathDisclosureScanRule extends PluginPassiveScanner {
+    private static final int PLUGIN_ID = 110009;
+    private static final String MESSAGE_PREFIX = "pscanalpha.fullpathdisclosurealert.";
+    private static final Logger LOGGER = LogManager.getLogger(FullPathDisclosureScanRule.class);
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(
+                    CommonAlertTag.OWASP_2021_A05_SEC_MISCONFIG,
+                    CommonAlertTag.WSTG_V42_ERRH_01_ERR);
+
+    @Override
+    public int getPluginId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (getHelper().isSuccess(msg)) {
+            return;
+        }
+        final String responseBody = msg.getResponseBody().toString();
+        String evidence;
+        // matches the following paths:
+        // example for windows : C:\folder\folder
+        // example for unix : /dir/dir/
+        // default paths : /usr or /home etc..
+        Pattern pathPattern =
+                Pattern.compile(
+                        ("(?:/bin/|/usr/|/mnt/|/proc/|/sbin/|/dev/|/lib/|/tmp/|/opt/|/home/|/var/|/root/|/etc/|"
+                                + "/Applications/|/Volumes/|/System/|/Users/|/Developer/|/Library/|"
+                                + "[a-z]\\:(\\\\Program Files\\\\|\\\\Users\\\\|\\\\Windows\\\\|\\\\ProgramData\\\\|\\\\Progra~1\\\\).*)"),
+                        Pattern.CASE_INSENSITIVE);
+
+        Matcher pathMatcher = pathPattern.matcher(responseBody);
+
+        if (pathMatcher.find()) {
+            evidence = pathMatcher.group();
+            LOGGER.debug("Found full path in response : {}", evidence);
+            buildAlert(evidence).raise();
+        }
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        return ALERT_TAGS;
+    }
+
+    public String getHelpLink() {
+        return "https://www.zaproxy.org/docs/desktop/addons/passive-scan-rules-alpha/#full-path-disclosure-scan-rule";
```

Comment:
Use the ID of the scan rule which should not change, e.g.
https://github.com/zaproxy/zap-extensions/pull/5092/files#diff-9ed1c472aeebc4831fb9eab55a03b04dbd7bcb324d21ea6bfe2bd5ac38b5e7e0R51
https://github.com/zaproxy/zap-extensions/pull/5092/files#diff-f3eb759ff0a547e9eeeaf7d75acdb28254152d66748c4d8d1353eee139e4d305R15

Reply:
For the code side, the class should just need to extend `CommonPassiveScanRuleInfo`.

Reply:
Might need to rebase your feature branch to pickup that interface.